### PR TITLE
Fix imports when same symbol is imported in different files

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/2032

### CODE REVIEW

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -3827,28 +3816,27 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
     }
 
     void visit_Module(const AST::Module_t &x) {
-        if (!current_scope) {
+        ASR::asr_t *tmp0 = nullptr;
+        if (current_scope) {
+            tmp0 = current_scope->asr_owner;
```
Comment (certik) [inline src/lpython/semantics/python_ast_to_asr.cpp:3821]: Put an assert here ensuring tmp0 is actually a TranslationUnit

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 2]
```diff
@@ -606,10 +596,10 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
     std::map<std::string, std::string> imported_functions;
 
     CommonVisitor(Allocator &al, LocationManager &lm, SymbolTable *symbol_table,
-            diag::Diagnostics &diagnostics, bool main_module,
+            diag::Diagnostics &diagnostics, bool main_module, std::string ext_mod_name,
```
[the remark this answers is not in the document]
Comment (Shaikh-Ubaid) [reply src/lpython/semantics/python_ast_to_asr.cpp]: Updated.

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 3]
```diff
@@ -3810,10 +3799,10 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
 
 
     SymbolTableVisitor(Allocator &al, LocationManager &lm, SymbolTable *symbol_table,
-        diag::Diagnostics &diagnostics, bool main_module,
+        diag::Diagnostics &diagnostics, bool main_module, std::string ext_mod_name,
```
[the remark this answers is not in the document]
Comment (Shaikh-Ubaid) [reply src/lpython/semantics/python_ast_to_asr.cpp]: Got it, documented.

File: src/lpython/semantics/python_ast_to_asr.cpp (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/lpython/semantics/python_ast_to_asr.cpp:3824]: Added.

**Comment by Shaikh-Ubaid** [comment]
Ready.

**Review by certik — APPROVED** [review]
I think that this looks good. Thanks @Shaikh-Ubaid !

**Pull request merged**